metrics/resource_view_test.go

```diff
@@ -341,19 +343,32 @@ testComponent_testing_value{project="p1",revision="r2"} 1
 			UnregisterResourceView(gaugeView, resourceCounter)
 
 			records := []metricExtract{}
-			for record := range ocFake.published {
-				for _, m := range record.Metrics {
-					if len(m.Timeseries) > 0 {
-						labels := map[string]string{}
-						if record.Resource != nil {
-							labels = record.Resource.Labels
+		loop:
+			for {
+				select {
+				case record := <-ocFake.published:
+					if record == nil {
+						continue loop
```

Comment:
I guess it's not important here?
```suggestion
						continue
```

Reply:
Actually, you need the label, otherwise the `continue` applies to the `select`, rather than to the loop.

Reply:
No, if it's a lambda you return from the func rather than continue. But since we're keeping the label, it doesn't matter.